pkg/solidity-utils/contracts/test/ERC4626TestToken.sol

```diff
@@ -39,6 +43,10 @@ contract ERC4626TestToken is ERC4626, IRateProvider {
         return _convertToAssets(FixedPoint.ONE, Math.Rounding.Floor);
     }
 
+    function mockRate(uint256 newRate) external {
+        _mockRate = newRate;
+    }
```

Comment:
Mocking rate like this is probably good enough for this test, but it does not cover rounding issues with the rate calculation itself. Not sure if it should be in this PR, but maybe we could create a mockRate that receives `assets` and `totalSupply`, minting underlying tokens and minting the wrapped itself. (minting underlying increases the rate, minting wrapped decreases the rate).